api/v1alpha1/clusterextension_types.go

```diff
@@ -370,48 +361,45 @@ type ServiceAccountReference struct {
 	//
 	// [RFC 1123]: https://tools.ietf.org/html/rfc1123
 	//
-	//+kubebuilder:validation:MaxLength:=253
-	//+kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
-	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
+	// +kubebuilder:validation:XValidation:rule="self.matches(\"^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$\")",message="name must be a valid DNS1123 subdomain"
+	// +kubebuilder:validation:Required
 	Name string `json:"name"`
 }
 
 // PreflightConfig holds the configuration for the preflight checks.  If used, at least one preflight check must be non-nil.
+//
 // +kubebuilder:validation:XValidation:rule="has(self.crdUpgradeSafety)",message="at least one of [crdUpgradeSafety] are required when preflight is specified"
 type PreflightConfig struct {
```

Comment:
We do not have optional tag here because we have `// +kubebuilder:validation:XValidation:rule="has(self.crdUpgradeSafety)"`?

Reply:
Correct.  There is a discussion in the other PR about how this `CRDUpgradeSafety` is a pointer, but also a required type, where we assert that we'll change this in the short term to make this an "at least one non-nil field" enforcement... and right now we only have the CRDUpgradeSafety type right now.